networking_cisco/plugins/cisco/cfg_agent/device_drivers/asr1k/aci_asr1k_routing_driver.py

```diff
@@ -186,13 +218,22 @@ def _remove_tenant_net_route(self, ri, port):
             out_itfc = self._get_interface_name_from_hosting_port(port)
             ip = netaddr.IPNetwork(cidr)
             subnet, mask = ip.network.format(), ip.netmask.format()
-            gateway_ip = self._get_interface_gateway_ip_from_hosting_port(port)
+            gateway_ip = self._get_interface_gateway_ip_from_hosting_port(
+                ri, port)
             conf_str = snippets.REMOVE_TENANT_ROUTE_WITH_INTF % (
                 vrf_name, subnet, mask, out_itfc, gateway_ip)
             self._edit_running_config(conf_str,
                                       'REMOVE_TENANT_ROUTE_WITH_INTF')
 
-    def _get_interface_ip_from_hosting_port(self, port, is_external=False):
+    def _get_info_port(self, ri, port):
```

Comment:
Right -- the intent here is to only return the interface port if there's not any form of gateway port available. This may be removed in the future, as we clean up what state gets passed down with what port in the plugging driver.